sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiverAsyncClient.java

```diff
@@ -113,13 +113,25 @@
         final ReceiverOptions newReceiverOptions = new ReceiverOptions(receiverOptions.getReceiveMode(),
             receiverOptions.getPrefetchCount(), receiverOptions.getMaxLockRenewDuration(),
             receiverOptions.isEnableAutoComplete(), sessionId, null);
+
         final ServiceBusSessionManager sessionSpecificManager = new ServiceBusSessionManager(entityPath, entityType,
             connectionProcessor, tracerProvider, messageSerializer, newReceiverOptions);
 
-        return sessionSpecificManager.getActiveLink().map(receiveLink -> new ServiceBusReceiverAsyncClient(
-            fullyQualifiedNamespace, entityPath, entityType, newReceiverOptions, connectionProcessor,
-            ServiceBusConstants.OPERATION_TIMEOUT, tracerProvider, messageSerializer, () -> { },
-            sessionSpecificManager));
+        if (CoreUtils.isNullOrEmpty(sessionId)) {
+            return sessionSpecificManager.getActiveLink().map(receiveLink -> new ServiceBusReceiverAsyncClient(
+                fullyQualifiedNamespace, entityPath, entityType, newReceiverOptions, connectionProcessor,
+                ServiceBusConstants.OPERATION_TIMEOUT, tracerProvider, messageSerializer, () -> {
+            }, sessionSpecificManager));
+        } else {
```

Comment:
sessionId can not be null or empty for this method. This branch isn't reachable at runtime?

Reply:
removed.